.github/workflows/docs.yml

```diff
@@ -24,91 +24,159 @@ concurrency:
         env:
           LD_LIBRARY_PATH: ${{ env.JAVA_HOME }}/lib/server:${{ env.LD_LIBRARY_PATH }}
+      
+      - name: Upload docs
+        uses: actions/upload-artifact@v2
+        with:
+          name: rust-docs
+          path: ./target/doc
 
-      - name: Copy docs into build
-        run: |
-          cp -r ./target/doc ./website/static/docs/rust
+  build-java-doc:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
 
-      # Build bindings/java docs --------------------------------------
+      - name: Setup Rust toolchain
+        uses: ./.github/actions/setup
+        with:
+          cache-key: "-services-all"
+      
+      - uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: '17'
+      
       - name: Build and test
         working-directory: bindings/java
         run: mvn javadoc:javadoc
-
-      - name: Copy docs into build
-        run: |
-          cp -r ./bindings/java/target/site/apidocs ./website/static/docs/java
-
-      # Set node.js environment -----------------------------------------
-      - uses: actions/setup-node@v3
+  
+      - name: Upload docs
+        uses: actions/upload-artifact@v2
         with:
-          node-version: '16'
-      - run: corepack enable
-        working-directory: bindings/nodejs
+          name: java-docs
+          path: ./bindings/java/target/site/apidocs
 
-      - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
-        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
+  build-nodejs-doc:
+    runs-on: ubuntu-latest
 
-      - uses: actions/cache@v3
-        id: yarn-cache
-        with:
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+    steps:
+      - uses: actions/checkout@v3
 
-      # Build bindings/nodejs docs --------------------------------------
       - name: Install dependencies
         working-directory: bindings/nodejs
         run: yarn install
-
+      
       - name: Build bindings/nodejs Docs
         working-directory: bindings/nodejs
         run: yarn docs
+      
+      - name: Upload docs
+        uses: actions/upload-artifact@v2
+        with:
+          name: nodejs-docs
+          path: ./bindings/nodejs/docs
 
-      - name: Copy docs into build
-        run: |
-          cp -r ./bindings/nodejs/docs ./website/static/docs/nodejs
+  build-python-doc:
+    runs-on: ubuntu-latest
 
-      # Setup python environment ----------------------------------------
-      - uses: actions/setup-python@v4
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v3
         with:
           python-version: '3.11'
-
-      # Build bindings/python docs --------------------------------------
+      
       - name: Install dependencies
         working-directory: bindings/python
         run: |
           set -e
           python -m pip install -e .
```

Comment:
This task requires a complete build of Python bindings with OpenDAL core. However, it's possible that this step may not be necessary. It seems that pdoc can generate documentation directly from the Python code. Would you like to give it a try?

Reply:
Could give it a try.